pkg/rulefmt/rulefmt.go

```diff
@@ -21,6 +21,7 @@ import (
 
 	"github.com/pkg/errors"
 	"github.com/prometheus/common/model"
+	yamlv2 "gopkg.in/yaml.v2"
```

Comment:
Why do we need two versions?

Reply:
We need strict unmarshalling checks which is only in yaml.v2 lib. Yamlv3 was required for line and columns support. Hence, I don't think we have any other way than using the features from both lib. Later if v3 gets strict unmarshalling, then this could be confined to a single lib. Other ways might be a hack and not recommended.

What do you think?

Reply:
I wasn't aware that this had regressed, which would be a bug anywhere we use v3 then. We either need to get it added to v3, return to v2, or use the approach we used before v2 had this.

Reply:
well, in that case, I will open an issue in yaml v3. Sticking to this will be good I think since if we switch over to v2, then features like column and line numbers would be lost.

Reply:
Do you know how this slipped past our unittests?

Reply:
Our unittests does not cover the criteria where an unknown field is present. Will update that in this PR as the yaml v3 is fixed.

Link of the issue in go-yaml v3: https://github.com/go-yaml/yaml/issues/602